### PR TITLE
Add grid to the doc start page

### PR DESCRIPTION
**Pull request by FedericoNegri**
image

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -23,3 +10,42 @@ PyHPS brings HPS to your Python app. Wrapping around HPS REST APIs, PyHPS allows
...
+
+.. grid:: 1 1 2 2
+        :gutter: 2
+
+        .. grid-item-card:: :octicon:`rocket` Getting started
+            :link: getting_started/index
+            :link-type: doc
+
+            Contains installation instructions to get you
+            started with PyHPS.
```
Comment (PipKat) [inline doc/source/index.rst:23]:
```suggestion
            Learn how to install and start using PyHPS.
```

File: doc/source/index.rst [hunk 2]
```diff
@@ -23,3 +10,42 @@ PyHPS brings HPS to your Python app. Wrapping around HPS REST APIs, PyHPS allows
...
+            :link: user_guide/index
+            :link-type: doc
+
+            Walks you through the basics of how to interact with HPS.
+
+        .. grid-item-card:: :octicon:`play` Examples
+            :link: examples/index
+            :link-type: doc
+
+            A collection of examples demonstrating how to submit jobs using PyHPS.
```
Comment (PipKat) [inline doc/source/index.rst:35]:
```suggestion
            Explore examples demonstrating how to use PyHPS to submit jobs.
```

File: doc/source/index.rst [hunk 3]
```diff
@@ -23,3 +10,42 @@ PyHPS brings HPS to your Python app. Wrapping around HPS REST APIs, PyHPS allows
...
+            :link-type: doc
+
+            Contains installation instructions to get you
+            started with PyHPS.
+
+        .. grid-item-card:: :octicon:`tools` User guide
+            :link: user_guide/index
+            :link-type: doc
+
+            Walks you through the basics of how to interact with HPS.
```
Comment (PipKat) [inline doc/source/index.rst:29]:
```suggestion
            Understand the basics of how to interact with HPS.
```

**Comment by PipKat** [comment]
@FedericoNegri To follow our standard content structure, you need to move the card for the "API reference" section before the card for the "Examples" section. (I already made a suggested edit to the toc in the overall ``index.rst`` file to do this reordering.) Also, you need to add a card to the end for the "Contribute" section:
``.. grid-item-card:: Contribute :fa:`people-group`
        :padding: 2 2 2 2
        :link: contributing
        :link-type: doc

        Learn how to contribute to the PyHPS codebase
        or documentation.

File: doc/source/index.rst [hunk 4]
```diff
@@ -23,3 +10,42 @@ PyHPS brings HPS to your Python app. Wrapping around HPS REST APIs, PyHPS allows
...
+            :link: examples/index
+            :link-type: doc
+
+            A collection of examples demonstrating how to submit jobs using PyHPS.
+
+        .. grid-item-card:: :octicon:`file-code` API reference
+            :link: api/index
+            :link-type: doc
+
+            Describes the public Python classes, methods, and functions.
```
[the remark this answers is not in the document]
Comment (FedericoNegri) [reply doc/source/index.rst]: I've take the freedom to rephrase it as 

> Understand the PyHPS API, its capabilities, and how to interact with it programmatically.

(it's unusual to talk about Python API endpoints, it's more a terminology for REST APIs)

**Comment by FedericoNegri** [comment]
> @FedericoNegri To follow our standard content structure, you need to move the card for the "API reference" section before the card for the "Examples" section. (I already made a suggested edit to the toc in the overall `index.rst` file to do this reordering.) Also, you need to add a card to the end for the "Contribute" section: ``.. grid-item-card:: Contribute :fa:`people-group` :padding: 2 2 2 2 🔗 contributing :link-type: doc
> 
> ```
>     Learn how to contribute to the PyHPS codebase
>     or documentation.
> ```

done, thanks

**Pull request merged**